src/Controller/ViewController.php

```diff
@@ -59,69 +38,60 @@ public function list(Request $request, ViewRepository $viewRepository): Response
     public function add(Request $request, ViewRepository $viewRepository): Response
     {
-        // Create session for multistep form.
-        try {
-            $session = $request->getSession();
-        } catch (SessionNotFoundException $e) {
-            $session = new Session();
-            $session->start();
-        }
-
-        $data = $session->get(self::VIEW_CREATE_SESSION_KEY);
-
-        // Initialize multistep form.
-        if (empty($data)) {
-            $data = $this->createFormInit();
-            $session->set(self::VIEW_CREATE_SESSION_KEY, $data);
-        } else {
-            // Re-initialize form if it's older than 10 minutes.
-            $created = clone $data['created'];
-            $created->modify('+10 minutes');
-            if ($created < new \DateTime()) {
-                $data = $this->createFormInit();
-                $session->set(self::VIEW_CREATE_SESSION_KEY, []);
-            }
-        }
-
-        $form = $this->createForm(self::STEPS[$data['current_step']]['class'], $data['view']);
+        $view = new View();
+        $view->setProtected(false);
+        $form = $this->createForm(ViewAddType::class, $view);
         $form->handleRequest($request);
 
-        if ($form->isSubmitted()) {
-            // If submitting on last step we save the data.
-            if (self::CREATE_FORM_LAST_STEP === $data['current_step']) {
-                if ($form->isValid()) {
-                    $session->set(self::VIEW_CREATE_SESSION_KEY, []);
-
-                    $viewRepository->save($data['view'], true);
+        if ($form->isSubmitted() && $form->isValid()) {
+            $viewRepository->save($view, true);
 
-                    return $this->redirectToRoute('app_view_list', $this->viewService->addView([
-                        'id' => $data['view']->getId(),
-                    ]), Response::HTTP_SEE_OTHER);
-                }
-            } else {
-                ++$data['current_step'];
-                $session->set(self::VIEW_CREATE_SESSION_KEY, $data);
-
-                return $this->redirectToRoute('app_view_add', $this->viewService->addView([]), Response::HTTP_SEE_OTHER);
-            }
+            return $this->redirectToRoute('app_view_edit', $this->viewService->addView([
+                'id' => $view->getId(),
+            ]), Response::HTTP_SEE_OTHER);
         }
 
-        return $this->render(self::STEPS[$data['current_step']]['template'], $this->viewService->addView([
+        return $this->render('view/add.html.twig', $this->viewService->addView([
             'form' => $form,
         ]));
     }
 
     #[Route('/{id}/edit', name: 'app_view_edit')]
-    public function edit(Request $request): Response
+    public function edit(Request $request, ViewRepository $viewRepository): Response
     {
-        return $this->redirectToRoute('app_view_list', $this->viewService->addView([]), Response::HTTP_SEE_OTHER);
-    }
+        $viewId = $request->attributes->get('id');
```

Comment:
No need for this.
The view can be autoextracted in the parameters by symfony.
```
public function edit(Request $request, View $view): Response
```

see https://github.com/itk-dev/economics/blob/develop/src/Controller/InvoiceController.php#L93. $invoice is auto-filled from the id